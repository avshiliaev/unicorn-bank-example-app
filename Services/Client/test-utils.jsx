import React from 'react';
import {render} from '@testing-library/react';
// this adds custom jest matchers from jest-dom
import '@testing-library/jest-dom/extend-expect';

export * from '@testing-library/react';
