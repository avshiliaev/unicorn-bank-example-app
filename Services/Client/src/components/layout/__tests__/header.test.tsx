import React from 'react';
import { shallow } from 'enzyme';
import { cleanup } from '../../../../test-utils';
import HeaderBasic from '../header.basic';
import toJson from 'enzyme-to-json';

describe('Header', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {

    const wrapper = shallow(
      <HeaderBasic
        windowSize={{ large: true }}
        slotLeft={(<div>Left</div>)}
        slotMiddle={(<div>Middle</div>)}
        slotRight={(<div>Right</div>)}
      />,
    );
    expect(wrapper.contains(<div>Left</div>)).toBeTruthy();
    expect(wrapper.contains(<div>Middle</div>)).toBeTruthy();
    expect(wrapper.contains(<div>Right</div>)).toBeTruthy();

    expect(toJson(wrapper)).toMatchSnapshot()

  });
});
