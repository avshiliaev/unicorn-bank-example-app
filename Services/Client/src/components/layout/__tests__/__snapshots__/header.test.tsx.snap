// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without error 1`] = `
<Header
  style={
    Object {
      "backgroundColor": "#fff",
      "boxShadow": "0 2px 4px 0 rgba(0, 0, 0, 0.2), 0 3px 10px 0 rgba(0, 0, 0, 0.19)",
      "height": false,
      "paddingLeft": 16,
      "paddingRight": 16,
      "zIndex": 999,
    }
  }
>
  <Row
    align="middle"
    gutter={
      Array [
        Object {
          "lg": 16,
          "md": 0,
          "sm": 0,
          "xs": 0,
        },
        0,
      ]
    }
  >
    <Col
      lg={2}
      md={4}
      sm={4}
      xl={2}
      xs={4}
      xxl={2}
    >
      <Row
        align="middle"
        gutter={
          Array [
            Object {
              "lg": 16,
              "md": 0,
              "sm": 0,
              "xs": 0,
            },
            0,
          ]
        }
        justify="start"
      >
        <Col>
          <div>
            Left
          </div>
        </Col>
      </Row>
    </Col>
    <Col
      lg={16}
      md={16}
      sm={16}
      xl={16}
      xs={16}
      xxl={16}
    >
      <Row
        align="middle"
        gutter={
          Array [
            Object {
              "lg": 16,
              "md": 0,
              "sm": 0,
              "xs": 0,
            },
            0,
          ]
        }
        justify="start"
      >
        <Col>
          <div>
            Middle
          </div>
        </Col>
      </Row>
    </Col>
    <Col
      lg={6}
      md={4}
      sm={4}
      xl={6}
      xs={4}
      xxl={6}
    >
      <Row
        align="middle"
        gutter={
          Array [
            Object {
              "lg": 16,
              "md": 0,
              "sm": 0,
              "xs": 0,
            },
            0,
          ]
        }
        justify="end"
      >
        <Col>
          <div>
            Right
          </div>
        </Col>
      </Row>
    </Col>
  </Row>
</Header>
`;
